app/gui/controller/double-representation/src/context_switch.rs

```diff
@@ -0,0 +1,214 @@
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// FQN for the [`Context::Output`].
+const OUTPUT: [&str; 5] = ["Standard", "Base", "Runtime", "Context", "Output"];
+/// FQN for the [`ContextSwitch::Enable`].
+const ENABLE: [&str; 4] = ["Standard", "Base", "Runtime", "with_enabled_context"];
+/// FQN for the [`ContextSwitch::Disable`].
+const DISABLE: [&str; 4] = ["Standard", "Base", "Runtime", "with_disabled_context"];
+
+// ===============================
+// === ContextSwitchExpression ===
+// ===============================
+
+// === ContextSwitch
+
+/// The two possible context switches.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub enum ContextSwitch {
+    Enable,
+    Disable,
+}
+
+impl<'a> TryFrom<QualifiedNameRef<'a>> for ContextSwitch {
+    type Error = ();
+
+    fn try_from(qualified_name: QualifiedNameRef<'a>) -> Result<Self, Self::Error> {
+        // Unwraps are safe, because this function is tested.
+        let enable_name = QualifiedName::from_all_segments(ENABLE).unwrap();
+        let disable_name = QualifiedName::from_all_segments(DISABLE).unwrap();
+
+        if qualified_name == enable_name {
+            Ok(ContextSwitch::Enable)
+        } else if qualified_name == disable_name {
+            Ok(ContextSwitch::Disable)
+        } else {
+            Err(())
+        }
+    }
+}
+
+// === Context ===
+
+/// Available execution contexts.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub enum Context {
+    Output,
+}
+
+impl<'a> TryFrom<QualifiedNameRef<'a>> for Context {
+    type Error = ();
+
+    fn try_from(qualified_name: QualifiedNameRef<'a>) -> Result<Self, Self::Error> {
+        // Unwrap is safe, because this function is tested.
+        let output_name = QualifiedName::from_all_segments(OUTPUT).unwrap();
+
+        if qualified_name == output_name {
+            Ok(Context::Output)
+        } else {
+            Err(())
+        }
+    }
+}
+
+// === Environment ===
+
+im_string_newtype! {
+    /// The name of the execution environment.
+    Environment
+}
+
+// === ContextSwitchExpression ===
+
+/// A representation of a single context switch expression.
+#[derive(Debug, Clone, PartialEq)]
+#[allow(missing_docs)]
+pub struct ContextSwitchExpression {
+    pub switch:      ContextSwitch,
+    pub context:     Context,
+    pub environment: Environment,
+}
+
+impl ContextSwitchExpression {
+    /// Parse the context switch expression from the given AST.
+    pub fn parse(ast: &Ast) -> Option<Self> {
+        let infix = known::Infix::try_new(ast.clone()).ok()?;
+        if ast::opr::is_right_assoc_opr(&infix.opr) {
+            let prefix = ast::prefix::Chain::from_ast(&infix.larg)?;
+            let infix_chain = ast::opr::Chain::try_new(&prefix.func)?;
+            let name_segments = infix_chain.as_qualified_name_segments()?;
+            let qualified_name = QualifiedName::from_all_segments(name_segments).ok()?;
```

Comment:
I think getting QualifiedName from AST or Infix chain could be a very common scenario. Consider adding a function for that somewhere (ast::infix or where QualifiedName is defined).

Reply:
Added TryFrom for QualifiedName